aidial_adapter_vertexai/chat/gemini/adapter.py

```diff
@@ -209,19 +225,17 @@ async def chat(
 
     @override
     async def count_prompt_tokens(self, prompt: GeminiPrompt) -> int:
-        # NOTE: Currently tools/functions couldn't be passed to the count_tokens method:
-        # https://github.com/googleapis/python-aiplatform/issues/3631
-        prompt.tools.not_supported()
-
         with Timer("count_tokens[prompt] timing: {time}", log.debug):
-            resp = await self.model.count_tokens_async(prompt.contents)
+            resp = await self._get_model(prompt=prompt).count_tokens_async(
```

Comment:
Why we're not including tools in this count, since this parameter was added to request?

https://github.com/googleapis/python-aiplatform/blob/3e7bf819d95167502be6b9f04c7b92e8415e5a68/vertexai/generative_models/_generative_models.py#L909

Reply:
Ah, that's done included from model default, nevermind